bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/api/CatalogQuery.scala

```diff
@@ -1,31 +1,30 @@
 package edu.gemini.catalog.api
 
-import edu.gemini.spModel.core.{Magnitude, Coordinates}
+import edu.gemini.spModel.core.Coordinates
 import edu.gemini.spModel.core.Target.SiderealTarget
 
 import scalaz._
 import Scalaz._
 
-sealed trait CatalogQuery {
-  val id: Option[Int] = None
-  val base: Coordinates
-  val radiusConstraint: RadiusConstraint
-  val catalog: CatalogName
+trait QueryResultsFilter {
+  def filter(t: SiderealTarget): Boolean
+}
+
+case class RadiusFilter(base: Coordinates, rc: RadiusConstraint) extends QueryResultsFilter {
+  def filter(t: SiderealTarget): Boolean = rc.targetsFilter(base)(t)
+}
 
-  def filter: SiderealTarget => Boolean = (t) => radiusConstraint.targetsFilter(base)(t) && magnitudeConstraints.map(_.filter(t)).getOrElse(true)
-  // TODO Move these filters out of CatalogQuery
-  val magnitudeConstraints: Option[MagnitudeConstraints]
-  val magnitudeRange: Option[MagnitudeRange]
-  def filterOnMagnitude(t: SiderealTarget, m:Magnitude): Boolean = radiusConstraint.targetsFilter(base)(t) && magnitudeConstraints.map(_.filter(t)).getOrElse(true) && magnitudeRange.map(_.filter(t, m)).getOrElse(true)
-  def filterOnMagnitude(t: SiderealTarget, m:Option[Magnitude]): Boolean =
-    radiusConstraint.targetsFilter(base)(t) && magnitudeConstraints.map(_.filter(t)).getOrElse(true) &&
-          m.isDefined && magnitudeRange.map(_.filter(t, m.get)).getOrElse(true)
-  def filterOnMagnitude(t: SiderealTarget, m:Option[Magnitude], rangeAdjustment: (Option[MagnitudeRange], Magnitude) => Option[MagnitudeRange]): Boolean =
-    radiusConstraint.targetsFilter(base)(t) && magnitudeConstraints.map(_.filter(t)).getOrElse(true) &&
-      m.isDefined && rangeAdjustment(magnitudeRange, m.get).map(_.filter(t, m.get)).getOrElse(true)
+case class MagnitudeQueryFilter(mc: MagnitudeConstraints) extends QueryResultsFilter {
+  def filter(t: SiderealTarget): Boolean = mc.filter(t)
+}
+
+/**
+ * Represents a query on a acatalog, indicating the location, radius and an optional set of magnitude constraints
+ */
+case class CatalogQuery(id: Option[Int], base: Coordinates, radiusConstraint: RadiusConstraint, magnitudeConstraints: List[MagnitudeConstraints], catalog: CatalogName) {
 
-  def withMagnitudeConstraints(magnitudeConstraints: Option[MagnitudeConstraints]): CatalogQuery = this
-  def withMagnitudeRange(magnitudeRange: Option[MagnitudeRange]): CatalogQuery = this
+  val filters: NonEmptyList[QueryResultsFilter] = NonEmptyList(RadiusFilter(base, radiusConstraint), magnitudeConstraints.map(MagnitudeQueryFilter.apply): _*)
+  def filter(t: SiderealTarget):Boolean = filters.list.forall(_.filter(t))
```

Comment:
What about removing the `QueryResultsFilter` hierarchy altogether since they are just predicates and only used internally?

````scala
val radiusFilter = radiusConstraint.targetsFilter(base)
def filter(t: SiderealTarget): Boolean =  radiusFilter(t) && magnitudeConstraints.filter(t)
````